core/src/main/java/hudson/model/Queue.java

```diff
@@ -2443,12 +2458,20 @@ protected NotWaitingItem(NotWaitingItem ni) {
      * {@link Item} in the {@link Queue#blockedProjects} stage.
      */
     public final class BlockedItem extends NotWaitingItem {
-        public BlockedItem(WaitingItem wi) {
```

Comment:
It was a public class. The old constructor has to be retained. `causeOfBlockage` is nullable, hence it should not be a problem

Reply:
@oleg-nenashev Can I mark old constructors as `@Deprecated` in this case?

Reply:
works for me